runtime/src/dpe_crypto.rs

```diff
@@ -169,42 +277,30 @@ impl<'a> Crypto for DpeCrypto<'a> {
         label: &[u8],
         info: &[u8],
     ) -> Result<(Self::PrivKey, EcdsaPub), CryptoError> {
-        match algs {
-            AlgLen::Bit256 => Err(CryptoError::Size),
-            AlgLen::Bit384 => {
-                hmac384_kdf(
-                    self.hmac384,
-                    KeyReadArgs::new(*cdi).into(),
-                    label,
-                    Some(info),
-                    self.trng,
-                    KeyWriteArgs::new(KEY_ID_TMP, KeyUsage::default().set_ecc_key_gen_seed_en())
-                        .into(),
-                )
-                .map_err(|e| CryptoError::CryptoLibError(u32::from(e)))?;
+        self.derive_key_pair_inner(algs, cdi, label, info, KEY_ID_DPE_PRIV_KEY)
+    }
 
-                let pub_key = self
-                    .ecc384
-                    .key_pair(
-                        &Ecc384Seed::Key(KeyReadArgs::new(KEY_ID_TMP)),
-                        &Array4x12::default(),
-                        self.trng,
-                        KeyWriteArgs::new(
-                            KEY_ID_DPE_PRIV_KEY,
-                            KeyUsage::default().set_ecc_private_key_en(),
-                        )
-                        .into(),
-                    )
-                    .map_err(|e| CryptoError::CryptoLibError(u32::from(e)))?;
-                let pub_key = EcdsaPub {
-                    x: CryptoBuf::new(&<[u8; AlgLen::Bit384.size()]>::from(pub_key.x))
-                        .map_err(|_| CryptoError::Size)?,
-                    y: CryptoBuf::new(&<[u8; AlgLen::Bit384.size()]>::from(pub_key.y))
-                        .map_err(|_| CryptoError::Size)?,
-                };
-                Ok((KEY_ID_DPE_PRIV_KEY, pub_key))
+    #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
+    fn derive_key_pair_exported(
+        &mut self,
+        algs: AlgLen,
+        exported_handle: &ExportedCdiHandle,
+        label: &[u8],
+        info: &[u8],
+    ) -> Result<(Self::PrivKey, EcdsaPub), CryptoError> {
+        let cdi = {
+            let mut cdi = None;
+            for cdi_slot in self.exported_cdi_slots.iter() {
+                match cdi_slot {
+                    Some((stored_cdi, stored_handle)) if stored_handle == exported_handle => {
+                        cdi = Some(*stored_cdi);
```

Comment:
Once the CDI is found should we break out of the loop?

Reply:
Good point, added!